### ISSUE TITLE
Automatically compile the storyboard before saving it if the sgl code changed

### ISSUE THREAD

**Issue by Damnae** [halbekanne/osu-sgl#23]
Because I always forget to press that button and wonder why my changes did nothing.

**Comment by halbekanne** [comment]
Is the GUI more clear now? :D
I will think about usability though.
![ss 2013-02-28 at 01 05 32 ](https://f.cloud.github.com/assets/692087/203528/c55a90aa-813a-11e2-82c5-76c2c5e7101a.jpg)

**Comment by halbekanne** [comment]
The problem with compiling before saving is that it's against the usual IDE convention where you would like to save your progress often but only compile when you think it's working. What will be possible is that when you click on the big blue play icon, it will compile and then save automatically. So that will be your all-in-one button ^^

**Comment by Damnae** [comment]
Yeah I meant to combine compiling the .osb + saving the .osb, not to compile when you save the .sgf.

That interface looks good (I guess it means the images could be dragged / double clicked to add the code automatically ?) because this way a new user isn't thrown in front of an empty text editor. But that image selector takes a bit too much space for how often you'll use it. I think it would be better if it could be reduced / closed.

**Comment by halbekanne** [comment]
> I guess it means the images could be dragged / double clicked to add the code automatically ?

Yes, if you double-click on a list entry, it will be added automatically.

> But that image selector takes a bit too much space for how often you'll use it. I think it would be better if it could be reduced / closed.

You can change the size, they are in a split pane. I will consider an option to reduce the needed space though. By the way are you good with GUI things in C#?

**Comment by Damnae** [comment]
> By the way are you good with GUI things in C#?

I've never really tried to use them...
